### PR TITLE
Remove duplicities in dependencies

### PR DESCRIPTION
**Pull request by richtja**
This creates a Dependency class to make dependencies hashable. Thanks to this change, we can easily find duplicates and remove them. This change is really necessary for Job dependencies, where the duplicates can be easily created by adding the same dependency to a test and job.

### CODE REVIEW

File: avocado/core/safeloader/docstring.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import json
 import re
 
+from avocado.plugins.dependency import Dependency
```
Comment (clebergnu) [inline avocado/core/safeloader/docstring.py:4]: `avocado.core` should not be importing from `avocado.plugins`, it should be the opposite.  The easy fix is to move `class Dependency` to, say, `avocado.core.nrunner.dependency`.

**Review by clebergnu — CHANGES_REQUESTED** [review]
Hi @richtja,

Thanks for this.  It works great, but I have put out a few requests and comments.

File: avocado/core/safeloader/docstring.py (hunk 1 above)
Comment (richtja) [reply avocado/core/safeloader/docstring.py:4]: Thank you for catching this, I was wondering why it was not caught by static-checks, but then I found out that we don't have such test. Therefore, I have created #5909

Comment (richtja) [reply avocado/core/safeloader/docstring.py:4]: Actually, I have looked it this more deeply, and I ponder if the safeloader should be in avocado.core. I understand that it is from historical reasons when avocado-instrumented test were part of legacy runner. But now the safeloader is used only by resolvers, and it is closely connected to them. Therefore, IMO it should be moved to plugins, but maybe there is something which I don't see. What do you think @clebergnu?

Comment (clebergnu) [reply avocado/core/safeloader/docstring.py:4]: I think the safeloader provides a generic core functionality, that is, the ability to parse Python code and find "stuff" without running them.  For now, I'd keep it where it is.

**Comment by richtja** [comment]
Hi @clebergnu, I have fixed the typos and move the `Dependency` class to `avocado.core.dependencies.dependency` please have a look.

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**